src/WorkflowManager/WorkflowManager/Validators/WorkflowValidator.cs

```diff
@@ -58,27 +71,26 @@ public static void Reset()
         /// - Unreferenced tasks other than root task.
         /// </summary>
         /// <param name="workflow">Workflow to validate.</param>
-        /// <param name="results">Workflow validation results.</param>
         /// <returns>if any validation errors are produced while validating workflow.</returns>
-        public static bool ValidateWorkflow(Workflow workflow, out (List<string> Errors, List<string> SuccessfulPaths) results)
+        public async Task<(List<string> Errors, List<string> SuccessfulPaths)> ValidateWorkflow(Workflow workflow)
         {
             workflow.IsValid(out var validationErrors);
             Errors.AddRange(validationErrors);
             var tasks = workflow.Tasks;
             var firstTask = tasks.FirstOrDefault();
 
-            ValidateWorkflowSpec(workflow);
+            await ValidateWorkflowSpec(workflow);
             DetectUnreferencedTasks(tasks, firstTask);
             ValidateTask(tasks, firstTask, 0);
             ValidateTaskDestinations(workflow);
             ValidateExportDestinations(workflow);
 
-            results = (Errors.ToList(), SuccessfulPaths.ToList());
+            var results = (Errors.ToList(), SuccessfulPaths.ToList());
```

Comment:
is the .ToList needed here?

Reply:
Yes it is, I removed this while testing something and forgot to add it back. Good spot, thank you.